include/bluetooth/mesh/scene_cli.h

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_scene_cli Scene Client model
+ * @{
+ * @brief API for the Scene Client model.
+ */
+
+#ifndef BT_MESH_SCENE_CLI_H__
+#define BT_MESH_SCENE_CLI_H__
+
+#include <bluetooth/mesh/scene.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct bt_mesh_scene_cli;
+
+/** @def BT_MESH_MODEL_SCENE_CLI
+ *
+ *  @brief Scene Client model composition data entry.
+ *
+ *  @param[in] _cli Pointer to a @ref bt_mesh_scene_cli instance.
+ */
+#define BT_MESH_MODEL_SCENE_CLI(_cli)                                          \
+	BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_SCENE_CLI, _bt_mesh_scene_cli_op,    \
+			 &(_cli)->pub,                                         \
+			 BT_MESH_MODEL_USER_DATA(struct bt_mesh_scene_cli,     \
+						 _cli),                        \
+			 &_bt_mesh_scene_cli_cb),
+
+/** Scene state */
+struct bt_mesh_scene_state {
+	/** Status of the previous operation. */
+	enum bt_mesh_scene_status status;
+	/** Current scene, or @ref BT_MESH_SCENE_NONE if no scene is active. */
+	uint16_t current;
+	/** Target scene, or @ref BT_MESH_SCENE_NONE if no transition is in
+	 *  progress.
+	 */
+	uint16_t target;
+	/** Remaining time of the scene transition in milliseconds. */
+	uint32_t remaining_time;
+};
+
+/** Scene register parameters */
+struct bt_mesh_scene_reg {
+	/** Status of the previous operation. */
+	enum bt_mesh_scene_status status;
+	/** Current scene. */
+	uint16_t current;
+	/** Number of entries in the scenes list. */
+	size_t count;
+	/** Optional list of scenes to populate. */
+	uint16_t *scenes;
+};
+
+/** Scene client model instance. */
+struct bt_mesh_scene_cli {
+	/** @brief Status message callback.
+	 *
+	 *  @param[in] cli   Scene client model.
+	 *  @param[in] ctx   Message context parameters.
+	 *  @param[in] state Received scene state.
+	 */
+	void (*status)(struct bt_mesh_scene_cli *cli,
+		       struct bt_mesh_msg_ctx *ctx,
+		       const struct bt_mesh_scene_state *state);
+
+	/** @brief Scene register message callback.
+	 *
+	 *  @param[in] cli Scene client model.
+	 *  @param[in] ctx Message context parameters.
+	 *  @param[in] reg Received scene register.
+	 */
+	void (*reg)(struct bt_mesh_scene_cli *cli, struct bt_mesh_msg_ctx *ctx,
+		    const struct bt_mesh_scene_reg *reg);
+
+	/* Composition data entry pointer. */
+	struct bt_mesh_model *mod;
+	/* Model publication parameters. */
+	struct bt_mesh_model_pub pub;
+	/* Publication message */
+	struct net_buf_simple pub_msg;
+	/* Publication message buffer. */
+	uint8_t buf[BT_MESH_MODEL_BUF_LEN(BT_MESH_SCENE_OP_RECALL,
+					  BT_MESH_SCENE_MSG_MAXLEN_RECALL)];
```

Comment:
I can't find a reason for doing this instead of initializing `pub->msg` and the handlers in a macro definition called something like `BT_MESH_SCENE_CLI_INIT`... Could you explain?

Reply:
Same for the server

Reply:
Same for the server

Reply:
NCSDK-6506. I initially planned on including this in the release, but there's no time. I'd rather not add to the problem now though.

The intent is to remove the need for as many of the INIT macros as possible, as they don't really serve any purpose except setting the publish message size for a lot of the models.

Reply:
Hmm, I don't see the problem in `INIT` macros, but I may not have enough context of the problem. Anyway it needs to be discussed separately.

If there are no other reasons, I think it would be better to stick to one approach, which is already established in the code, and to not create a mixed API. `INIT` macros can be removed later all at once.